ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java

```diff
@@ -428,21 +431,26 @@ private static void addAdditionalTransactionInformationToFlatTrace(
         .transactionPosition(
             block.getBody().getTransactions().indexOf(transactionTrace.getTransaction()))
         .transactionHash(transactionTrace.getTransaction().getHash().toHexString());
-    // add creationMethod for create action
-    Optional.ofNullable(builder.getType())
-        .filter(type -> type.equals("create"))
-        .ifPresent(__ -> addContractCreationMethodToFlatTrace(transactionTrace, builder));
+
+    addContractCreationMethodToTrace(transactionTrace, builder);
   }
 
-  private static void addContractCreationMethodToFlatTrace(
+  private static void addContractCreationMethodToTrace(
       final TransactionTrace transactionTrace, final FlatTrace.Builder builder) {
-    final String creationMethod =
-        transactionTrace.getTraceFrames().stream()
-            .filter(frame -> "CREATE2".equals(frame.getOpcode()))
-            .findFirst()
-            .map(TraceFrame::getOpcode)
-            .orElse("CREATE")
-            .toLowerCase(Locale.US);
-    builder.getActionBuilder().creationMethod(creationMethod);
+
+    // add creationMethod for create action
+    Optional.ofNullable(builder.getType())
+        .filter(type -> type.equals("create"))
+        .ifPresent(
+            __ -> {
+              final String creationMethod =
```

Comment:
Why using a local variable there ? You could pass directly this value to the builder

Reply:
Done 👍

Reply:
Nice